agent/src/beerocks/slave/ap_manager_thread.cpp

```diff
@@ -808,9 +808,12 @@ bool ap_manager_thread::handle_cmdu(Socket *sd, ieee1905_1::CmduMessageRx &cmdu_
             return true;
         }
 
-        LOG(DEBUG) << "CLIENT_ALLOW: mac = " << sta_mac << ", bssid = " << bssid;
+        // remove client from disallowed list
```

Comment:
The point is not to disagree... The idea is to tell a story with the comments. IMO one should be able to understand the intent of the code without reading the code itself, just by reading the comments. And yes, this approach sometimes results in stating the obvious. But sometimes few lines of code are doing something that is summed up in a single comment. When functionality is not trivial this 'story' expressed in comments is very useful for later maintenance.
For example, it is not uncommon for me to write a code like
`// do we have enough?`
`if ( enough ){`
`// it seems that we are good, continue...`
`}`
`else {`
`// we don't have enough, lets try another approach`
`}`
In this specific case that we are chatting on: the comment is completely redundant and should be removed as there are no other comments inside this case clause that supports this paradigm.

Reply:
No, I think that's wrong. We're programmers, we know how to read code (so long as it's written sanely, of course).
Comments should describe what happens at a higher layer than can be seen in the code directly. As such, they should only describe overall function behavior and how that function is used, rather than what it does. (i.e. instead of having a bunch of "remove item from array"-type comments within the function, the function should just have a description like "disassociates agent from the controller and removes its credentials so that it has to pair again").
Comments within functions  should be avoided, and only used to explain especially tricky, non-intuitive code (which should also be avoided if possible, of course).

"Telling a story with comments" just means the code is now telling two stories, and woe betide you if you let them get out of sync

Reply:
I agree, I don't think we have an argue here. I don't comment trivial code that can be read as is. As you said, there is  no point in doing so. However, sometimes the code is not trivial even if it doesn't do tricky things. It just has its logic that might be berried in few lines of technical details that have nothing to do with the logic itself. In these cases I find it extremely maintainable to comment the different sections in the function and their logic.
A small example can be seen in the place this discussion was started: https://github.com/prplfoundation/prplMesh/blob/master/agent/src/beerocks/slave/son_slave_thread.cpp#L4373-L4405
This else clause starts with a story about what is done in it. In this case not what would you expect. It also states why it is fine and suggest future changes.
Later in the code the promise that was taken in the initial comment is implemented.
I find it useful to read the first comment, then look forward in the code for _more_ comments that support the story on top, and only then dive into the code to ensure it does what it says it does.

All in all commenting code is a very delicate process that its benefits are revealed only *over time*. Comments are not compiled therefore there target is not correct code. They serve as another mean for maintaining a code base.
One indication for poor commenting process is that developers don't read comments. It means that they get used to comments that are irrelevant, hard to read or just outdated. This situation is in many cases worse then non commented code at all, because developers, seeking for assistance are reading misleading comments.

Reply:
Bottom line - comments in **prplMesh** are saying what **isn't obvious** from the code, and contributors are expected to align to this approach.

Reply:
I think @RanRegev's long remark is very valuable and it should be put in a new "Comments" section in CODINGSTYLE.md. @RanRegev can you do that?

Reply:
#1062